### PR TITLE
Indicate defaults in comandline help

### LINKED ISSUE #12682: Enhancement request: show default settings of command-line flags
```
seth@schroeder:~/dev/julia/wip/LightGraphs.jl$ julia --help
julia [switches] -- [programfile] [args...]
 -v, --version             Display version information
 -h, --help                Print this message

 -J, --sysimage <file>     Start up with the given system image file
 --precompiled={yes|no}    Use precompiled code from system image if available
 -H, --home <dir>          Set location of julia executable
 --startup-file={yes|no}   Load ~/.juliarc.jl
 -f, --no-startup          Don't load ~/.juliarc (deprecated, use --startup-file=no)
 -F                        Load ~/.juliarc (deprecated, use --startup-file=yes)
 --handle-signals={yes|no} Enable or disable Julia's default signal handlers

 -e, --eval <expr>         Evaluate <expr>
 -E, --print <expr>        Evaluate and show <expr>
 -P, --post-boot <expr>    Evaluate <expr>, but don't disable interactive mode (deprecated, use -i -e instead)
 -L, --load <file>         Load <file> immediately on all processors

 -p, --procs {N|auto}      Integer value N launches N additional local worker processes
                           "auto" launches as many workers as the number of local cores
 --machinefile <file>      Run processes on hosts listed in <file>

 -i                        Interactive mode; REPL runs and isinteractive() is true
 -q, --quiet               Quiet startup (no banner)
 --color={yes|no}          Enable or disable color text
 --history-file={yes|no}   Load or save history
 --no-history-file         Don't load history file (deprecated, use --history-file=no)

 --compile={yes|no|all}    Enable or disable compiler, or request exhaustive compilation
 -C, --cpu-target <target> Limit usage of cpu features up to <target>
 -O, --optimize            Run time-intensive code optimizations
 --inline={yes|no}         Control whether inlining is permitted (overrides functions declared as @inline)
 --check-bounds={yes|no}   Emit bounds checks always or never (ignoring declarations)
 --math-mode={ieee,fast}   Disallow or enable unsafe floating point optimizations (overrides @fastmath declaration)

 --depwarn={yes|no|error}  Enable or disable syntax and method deprecation warnings ("error" turns warnings into errors)

 --output-o name           Generate an object file (including system image data)
 --output-ji name          Generate a system image data file (.ji)
 --output-bc name          Generate LLVM bitcode (.bc)

 --output-incremental=no   Generate an incremental output file (rather than complete)

 --code-coverage={none|user|all}, --code-coverage
                           Count executions of source lines (omitting setting is equivalent to "user")
 --track-allocation={none|user|all}, --track-allocation
                           Count bytes allocated by each source line
```

It would be helpful to know what the default settings for each flag are, if defaults exist. (Or is it that the first option is always the default? If so, that would be useful information in the help text.)

### LINKED ISSUE #36017: Show --compile with --help (and support better)
A. Since `--compile=min` works (even as of 1.6, and I guess all versions since 1.0) and can be MUCH faster (for loading code), show it!

Is there a reason it's undocumented? Are some of of the options dangerous, or other reason undocumented? I believe they are "yes", "no", and "min", and if I recall for pre-1.0 "min" could crash?

B. As with Jeff's new selective lowered optimization trick, being able to get the above option that way would be nice. I'm not sure if that's an easy change, while even I could make a PR for A. unless it's not wanted.

C. Letting the user choose for the new opt option and/or possible B. would be nice, as e.g. `@lowopt using Module` and even better if possible more granularity, i.e. for functions (loops?).

### LINKED ISSUE #43949: startup.jl location
All the documentation, the manual and the `--help` options indicates that the `startup.jl` file is loaded from location `~/.julia/config/startup.jl`, while in fact it is influenced by the `JULIA_DEPOT_PATH` environment variable. I believe the documentation should be updated, as it is quite surprising. While I understand that it would make the documentation more heavy, it should at least appear when running `julia --help` with the right path.

### PR DESCRIPTION
**Pull request by laborg**
Fixes https://github.com/JuliaLang/julia/issues/12682
Fixes https://github.com/JuliaLang/julia/issues/43949
Fixes https://github.com/JuliaLang/julia/issues/36017

- `*` indicates the default if the options is unspecified
- Fixed a couple of inconsistencies (e.g. use ` instead of ")
- The man page and the command-line-options.md have been updated too

```
$ ./julia --help

    julia [switches] -- [programfile] [args...]

Switches (a '*' marks the default value, if applicable):

 -v, --version              Display version information
 -h, --help                 Print this message (--help-hidden for more)
 --help-hidden              Uncommon options not shown by `-h`

 --project[={<dir>|@.}]     Set <dir> as the home project/environment
 -J, --sysimage <file>      Start up with the given system image file
 -H, --home <dir>           Set location of `julia` executable
 --startup-file={yes*|no}   Load `JULIA_DEPOT_PATH/config/startup.jl`; if `JULIA_DEPOT_PATH`
                            environment variable is unset, load `~/.julia/config/startup.jl`
 --handle-signals={yes*|no} Enable or disable Julia's default signal handlers
 --sysimage-native-code={yes*|no}
                            Use native code from system image if available
 --compiled-modules={yes*|no}
                            Enable or disable incremental precompilation of modules

 -e, --eval <expr>          Evaluate <expr>
 -E, --print <expr>         Evaluate <expr> and display the result
 -L, --load <file>          Load <file> immediately on all processors

 -t, --threads {N|auto}     Enable N threads; "auto" tries to infer a useful default number
                            of threads to use but the exact behavior might change in the future.
                            Currently, "auto" uses the number of CPUs assigned to this julia
                            process based on the OS-specific affinity assignment interface, if
                            supported (Linux and Windows). If this is not supported (macOS) or
                            process affinity is not configured, it uses the number of CPU
                            threads.
 -p, --procs {N|auto}       Integer value N launches N additional local worker processes
                            "auto" launches as many workers as the number of local CPU threads (logical cores)
 --machine-file <file>      Run processes on hosts listed in <file>

 -i                         Interactive mode; REPL runs and `isinteractive()` is true
 -q, --quiet                Quiet startup: no banner, suppress REPL warnings
 --banner={yes|no|auto*}    Enable or disable startup banner
 --color={yes|no|auto*}     Enable or disable color text
 --history-file={yes*|no}   Load or save history

 --depwarn={yes|no*|error}  Enable or disable syntax and method deprecation warnings (`error` turns warnings into errors)
 --warn-overwrite={yes|no*} Enable or disable method overwrite warnings
 --warn-scope={yes*|no}     Enable or disable warning for ambiguous top-level scope

 -C, --cpu-target <target>  Limit usage of CPU features up to <target>; set to `help` to see the available options
 -O, --optimize={0,1,2*,3}  Set the optimization level (level 3 if `-O` is used without a level)
 --min-optlevel={0*,1,2,3}  Set a lower bound on the optimization level
 -g [{0,1*,2}]              Set the level of debug info generation (level 2 if `-g` is used without a level)
 --inline={yes*|no}         Control whether inlining is permitted, including overriding @inline declarations
 --check-bounds={yes|no|auto*}
                            Emit bounds checks always, never, or respect @inbounds declarations
 --math-mode={ieee,fast}    Disallow or enable unsafe floating point optimizations (overrides @fastmath declaration)

 --code-coverage[={none*|user|all}]
                            Count executions of source lines (omitting setting is equivalent to `user`)
 --code-coverage=tracefile.info
                            Append coverage information to the LCOV tracefile (filename supports format tokens)
 --track-allocation[={none*|user|all}]
                            Count bytes allocated by each source line (omitting setting is equivalent to `user`)
 --bug-report=KIND          Launch a bug report session. It can be used to start a REPL, run a script, or evaluate
                            expressions. It first tries to use BugReporting.jl installed in current environment and
                            fallbacks to the latest compatible BugReporting.jl if not. For more information, see
                            --bug-report=help.
```

```
$ ./julia --help-hidden

    julia [switches] -- [programfile] [args...]

Switches (a '*' marks the default value, if applicable):

 --compile={yes*|no|all|min}
                          Enable or disable JIT compiler, or request exhaustive or minimal compilation

 --output-o <name>        Generate an object file (including system image data)
 --output-ji <name>       Generate a system image data file (.ji)
 --strip-metadata         Remove docstrings and source location info from system image
 --strip-ir               Remove IR (intermediate representation) of compiled functions

 --output-unopt-bc <name> Generate unoptimized LLVM bitcode (.bc)
 --output-bc <name>       Generate LLVM bitcode (.bc)
 --output-asm <name>      Generate an assembly file (.s)
 --output-incremental={yes|no*}
                          Generate an incremental output file (rather than complete)
 --trace-compile={stderr,name}
                          Print precompile statements for methods compiled during execution or save to a path
 --image-codegen          Force generate code in imaging mode
```

### CODE REVIEW

**Comment by vtjnash** [comment]
still draft, or ready for review now?

**Comment by laborg** [comment]
Its ready for review. Not sure if the CI failures should tell me something.

File: doc/man/julia.1 [hunk 1]
```diff
@@ -112,77 +119,96 @@ process affinity is not configured, it uses the number of CPU
...
 .TP
 --math-mode={ieee|user}
-Always use IEEE semantics for math (ignoring declarations),
-or adhere to declarations in source code
+Disallow or enable unsafe floating point optimizations (overrides @fastmath declaration)
 
 .TP
---depwarn={yes|no|error}
-Enable or disable syntax and method deprecation warnings ('error' turns warnings into errors)
+--code-coverage=[{none*|user|all}]
```
Comment (vtjnash) [inline doc/man/julia.1:191]:
```suggestion
--code-coverage[={none*|user|all}]
```

File: doc/src/manual/command-line-options.md [hunk 2]
```diff
@@ -94,23 +94,34 @@ The following is a complete list of command-line switches available when launchi
...
+|`--check-bounds={yes\|no\|auto*}`      |Emit bounds checks always, never, or respect `@inbounds` declarations|
 |`--math-mode={ieee,fast}`              |Disallow or enable unsafe floating point optimizations (overrides `@fastmath` declaration)|
-|`--code-coverage={none\|user\|all}`    |Count executions of source lines|
-|`--code-coverage`                      |equivalent to `--code-coverage=user`|
-|`--track-allocation={none\|user\|all}` |Count bytes allocated by each source line|
-|`--track-allocation`                   |equivalent to `--track-allocation=user`|
-
+|`--code-coverage=[{none*\|user\|all}]` |Count executions of source lines (omitting setting is equivalent to `user`)|
+|`--code-coverage=tracefile.info`       |Append coverage information to the LCOV tracefile (filename supports format tokens).|
+|`--track-allocation=[{none*\|user\|all}]` |Count bytes allocated by each source line (omitting setting is equivalent to "user")|
```
Comment (vtjnash) [inline doc/src/manual/command-line-options.md:112]:
```suggestion
|`--code-coverage[={none*\|user\|all}]` |Count executions of source lines (omitting setting is equivalent to `user`)|
|`--code-coverage=tracefile.info`       |Append coverage information to the LCOV tracefile (filename supports format tokens).|
|`--track-allocation[={none*\|user\|all}]` |Count bytes allocated by each source line (omitting setting is equivalent to "user")|
```

**Comment by IanButterworth** [comment]
Marking this for 1.8 backport as I don't see any reason not to, and at least one PR for backport is on top of this

**Pull request merged**